samples/test/metrics_visualization_v2.py

```diff
@@ -23,8 +24,8 @@
 @component(
     packages_to_install=['sklearn'],
     base_image='python:3.9',
+    kfp_package_path=os.getenv('KFP_PACKAGE_PATH'),
```

Comment:
we plan to add tests for every sample in `samples/core` too (many of them already have tests).
search for "core" in https://github.com/kubeflow/pipelines/blob/master/samples/test/config.yaml

> Actually, this is per-component, not per-pipeline. It shouldn't be part of compiler options.

Why is this per-component? Shouldn't every component use the same runtime spec matching the sdk's version?

Reply:
No, each component should use the runtime version at the time the component was compiled, not when the pipeline was compiled. When we use the `@component` decorator, we're creating a shareable, static version of the component. That component's code should not change depending on when the pipeline was compiled.

Reply:
> When we use the @component decorator, we're creating a shareable, static version of the component. That component's code should not change depending on when the pipeline was compiled.

I'm a bit confused, IIUC in this PR the default behavior is to use current KFP SDK version. Because usually (from my understanding) people do not compile their lightweight components separately, so all the components in the pipeline will use the KFP SDK version at the time of pipeline compilation. That seems to contradict your statement. Are you assuming pre-compiling a lightweight component to component yaml and then use it in a pipeline should be what people do? What are pros and cons?

My gut feeling is that, because kfp v2 compatible launcher binary version is coupled to sdk version, so it's better to use the same version for container runtime too. Why would people prefer otherwise?

Besides that, as I mentioned above, we already run many core samples in v2 sample test infra.
Is it OK to use the default value for core samples and not add the env var?

Reply:
No, it's not ok to do that. Like I said, components are meant to be standalone, static versions of the code at the time the component was compiled, not when the pipeline was compiled. Component authoring can happen independently of pipeline authoring. Users can export component.yaml from lightweight components, which can be included at anytime, anywhere in a pipeline compiled with a completely different KFP version.